kad/kad.go

```diff
@@ -28,3 +28,40 @@ type Key[K any] interface {
 	// and 0 if both keys are equal.
 	Compare(other K) int
 }
+
+// RoutingTable is the interface all Kademlia Routing Tables types support.
+type RoutingTable[K Key[K]] interface {
+	// Self returns the local node's Kademlia key
+	Self() K
```

Comment:
Would it simplify/complicate things if we required `Self()` to return `NodeID[K]` instead of `K`?

Reply:
removed it for now. I only found one real usage, that can be done differently.